runtime/connections.go

```diff
@@ -65,3 +65,15 @@ func (r *Runtime) Catalog(ctx context.Context, instanceID string) (*catalog.Serv
 
 	return c, err
 }
+
+func (r *Runtime) Close() error {
+	c := r.connCache
+	for _, key := range c.cache.Keys() {
+		val, _ := c.cache.Get(key)
+		err := val.(drivers.Connection).Close()
+		if err != nil {
```

Comment:
When do we actually get error during close? Does it make sense to not close the remaining connections when one errors out?

Reply:
For example, the underlying DB driver could return an error during close. But I agree, we should not return immediately. @rakeshsharma14317, how about we a) log the error immediately, b) continue closing all other connections, and c) return the *first* encountered error, if any

Reply:
Also, I think this should be implemented as a `close` method on the `connectionCache` struct. And to be safe, it should use the lock to prevent race conditions, and track a boolean `closed` to ensure no connections are opened after it has been closed.